### PR TITLE
SDK-252: Updating settings.xml

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/SDK-252

 cc @ibacher ,@dkayiwa

### CODE REVIEW

File: maven-plugin/src/main/resources/settings.xml [hunk 1]
```diff
@@ -24,9 +24,17 @@
...
                     <name>OpenMRS Nexus Repository</name>
                     <url>http://mavenrepo.openmrs.org/nexus/content/repositories/public</url>
+                     <releases>
+				     <enabled>true</enabled>
+					 <checksumPolicy>fail</checksumPolicy>
+					 </releases>
+					 <snapshots>
+					 <enabled>true</enabled>
+					 <checksumPolicy>warn</checksumPolicy>
+					 </snapshots>
```
Comment (ibacher) [inline maven-plugin/src/main/resources/settings.xml:37]: Can you please ensure to consistently only use spaces for the formatting here? Also, we shouldn't remove the `openmrs-repo` identifier nor add the policies to it. We should instead create a new repository entry, even though it's referring to the same repository. This allows us to have a specific policy that applies to downloading archetypes, but not necessarily to other Maven artefacts.

File: pom.xml [hunk 2]
```diff
@@ -163,7 +163,7 @@
             <dependency>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-archetype-plugin</artifactId>
-                <version>2.4</version>
+                <version>3.0.0</version>
```
Comment (ibacher) [inline pom.xml:166]: On reflection, I wouldn't change this version right now. There is some brittle code elsewhere in the SDK, which is why the Travis build is failing.

File: pom.xml [hunk 3]
```diff
@@ -186,6 +186,11 @@
                 <artifactId>httpclient</artifactId>
                 <version>4.5.2</version>
             </dependency>
+           
+            <dependency>
+			    <groupId>org.dom4j</groupId>
+			    <artifactId>dom4j</artifactId>
+           </dependency>
```
Comment (ibacher) [inline pom.xml:193]: This is fine to keep in, but please consistently use on spaces for indentation here.

Comment (sherrif10) [reply pom.xml]: i think it looks ok now

File: maven-plugin/src/main/resources/settings.xml [hunk 4]
```diff
@@ -24,10 +24,22 @@
             </properties>
             <repositories>
                 <repository>
-                    <id>openmrs-repo</id>
+                    <id>openmrs repo</id>
```
Comment (ibacher) [inline maven-plugin/src/main/resources/settings.xml:27]: This should still be `openmrs-repo`

Comment (sherrif10) [reply maven-plugin/src/main/resources/settings.xml]: Sorry @ibacher   for late reply my computer had got issues but am now settled thanks for this

**Pull request merged**